modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/DefaultExecutionMediator.java

```diff
@@ -89,13 +91,14 @@ public DefaultExecutionMediator(ExtensionModel extensionModel, OperationModel op
    * @throws Exception if the operation or a {@link Interceptor#before(ExecutionContext)} invokation fails
    */
   @Override
-  public Mono<Object> execute(OperationExecutor executor, ExecutionContextAdapter context) throws MuleException {
+  public Publisher<Object> execute(OperationExecutor executor, ExecutionContextAdapter context) throws MuleException {
     final List<Interceptor> interceptors = collectInterceptors(context.getConfiguration(), executor);
     final Optional<MutableConfigurationStats> stats = getMutableConfigurationStats(context);
     stats.ifPresent(s -> s.addInflightOperation());
 
     try {
-      return (Mono<Object>) getExecutionTemplate(context).execute(() -> executeWithInterceptors(executor, context, interceptors, stats));
+      return (Mono<Object>) getExecutionTemplate(context)
```

Comment:
Yes. this PR is long enough I don't want to add changes on the execution template just for avoiding a cast.